tools/lib/tests/test_caching.py

```diff
@@ -34,6 +35,32 @@ def do_HEAD(self):
 
 class TestFileDownload(unittest.TestCase):
 
+  def test_defaults(self):
+    # TODO: parameterize the defaults so we don't rely on hard-coded values in xx
+
+    self.assertEqual(URLFile.pool_manager().pools._maxsize, 10) # PoolManager num_pools param
+    pool_manager_defaults = {
+      "maxsize": 100,
+      "socket_options": [(socket.SOL_SOCKET, socket.SO_KEEPALIVE, 1),],
+    }
+    for k, v in pool_manager_defaults.items():
+      self.assertEqual(URLFile.pool_manager().connection_pool_kw.get(k), v)
+
+    retry_defaults = {
+      "total": 5,
+      "backoff_factor": 0.5,
+      "status_forcelist": [409, 429, 503, 504],
+    }
+    for k, v in retry_defaults.items():
+      self.assertEqual(getattr(URLFile.pool_manager().connection_pool_kw["retries"], k), v)
+
+    try:
+      del os.environ["FILEREADER_CACHE"]
+    except KeyError:
+      pass
```

Comment:
not sure why this is set. our tests should always have a clean environment. @jnewb1 you know what's going on here?

if we have to do this, this is cleaner
```suggestion
    os.environ.pop("FILEREADER_CACHE", None)
```

Reply:
[this](https://github.com/commaai/openpilot/blob/master/conftest.py#L29-L34) just ensures that any environment variables changed within a single test don't get propagated to the next test. I think pytest uses some environment variables so I don't think we can clear them totally, but there might be a better way to handle this

Reply:
FILEREADER_CACHE is also set in CI:
https://github.com/commaai/openpilot/blob/master/.github/workflows/selfdrive_tests.yaml#L22